sdk/resourcemanager/azure-resourcemanager-compute/src/main/java/com/azure/resourcemanager/compute/models/DiskEncryptionSet.java

```diff
@@ -0,0 +1,236 @@
+             */
+            WithCreate withSystemAssignedIdentityBasedAccessToCurrentKeyVault(BuiltInRole builtInRole);
+        }
+
+        /**
+         * The stage of a disk encryption set definition allowing to enable automatic key rotation.
+         */
+        interface WithAutomaticKeyRotation {
+            /**
+             * Enable automatic key rotation.
+             * If enabled, the system will automatically update all managed disks, snapshots, and images
+             * referencing the disk encryption set to use the new version of the key within one hour.
+             * @return the next stage of the definition
+             */
+            WithCreate withAutomaticKeyRotation();
+        }
+
+        /**
+         * The stage of the definition which contains all the minimum required inputs for the resource to be created,
+         * but also allows for any other optional settings to be specified.
+         */
+        interface WithCreate
+            extends Creatable<DiskEncryptionSet>,
+                DefinitionWithTags<WithCreate>,
+                WithAutomaticKeyRotation { }
+    }
+
+    /** The template for an update operation, containing all the settings that can be modified. */
+    interface Update
+        extends Appliable<DiskEncryptionSet>,
+        UpdateStages.WithSystemAssignedManagedServiceIdentity,
+        UpdateStages.WithSystemAssignedIdentityBasedAccess,
+        UpdateStages.WithKeyVault,
+        UpdateStages.WithAutomaticKeyRotation {
+    }
+
+    /** Grouping of disk encryption set update stages. */
+    interface UpdateStages {
+        /**
+         * The stage of a disk encryption set update allowing to enable System Assigned (Local) Managed Service
+         * Identity.
+         */
+        interface WithSystemAssignedManagedServiceIdentity {
+            /**
+             * Specifies that System Assigned (Local) Managed Service Identity needs to be enabled in the disk
+             * encryption set.
+             * @return the next stage of the update
+             */
+            WithSystemAssignedIdentityBasedAccess withSystemAssignedManagedServiceIdentity();
+
+            /**
+             * Specifies that System Assigned (Local) Managed Service Identity needs to be disabled in the disk
+             * encryption set.
+             * @return the next stage of the update
+             */
+            Update withoutSystemAssignedManagedServiceIdentity();
+        }
+
+        /**
+         * The stage of a disk encryption set update allowing to associate with an Azure key vault.
+         */
+        interface WithKeyVault {
+            /**
+             * Associates with the disk encryption set an Azure key vault by its resource ID.
+             * @param keyVaultId resource ID of the Azure key vault
+             * @return the next stage of the update
+             */
+            WithKeyVaultKey withExistingKeyVault(String keyVaultId);
```

Comment:
I am not sure whether service would allow changing the vault. Please have it in test.

Reply:
OK, in portal they do, I'll add tests.

Reply:
Although they do support changing the key vault, they need the managed identity associated with the DES to have access to the new key vault(whether through RBAC or Access Policy) before applying updates to the DES. 

I can't find an easy way to support this, so I decide to remove the change of vault and keys altogether...